From d4c5a3da694e8293b8fc187290eba27d51523a1e Mon Sep 17 00:00:00 2001
From: Jeremy Drake <github@jdrake.com>
Date: Thu, 24 Sep 2020 12:20:12 -0700
Subject: [PATCH 8/8] Allow creating an empty import library for a dll.

After preventing creating an import library for an exe when there are no
exports, to avoid a crash, it turned out that some projects expected to
be able to create an import library for a dll with no exports, so more
closely match the condition to the condition around initializing the dll
name.
---
 ld/emultempl/pe.em  | 3 ++-
 ld/emultempl/pep.em | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/ld/emultempl/pe.em b/ld/emultempl/pe.em
index 32cd175ebb..6c14af5e29 100644
--- a/ld/emultempl/pe.em
+++ b/ld/emultempl/pe.em
@@ -2013,7 +2013,8 @@ gld_${EMULATION_NAME}_finish (void)
     {
       pe_dll_fill_sections (link_info.output_bfd, &link_info);
       if (command_line.out_implib_filename
-          && pe_def_file->num_exports != 0)
+          && (pe_def_file->num_exports != 0
+              || bfd_link_pic (&link_info)))
 	pe_dll_generate_implib (pe_def_file, command_line.out_implib_filename,
 				&link_info);
     }
diff --git a/ld/emultempl/pep.em b/ld/emultempl/pep.em
index 0b43ec515d..0918a68059 100644
--- a/ld/emultempl/pep.em
+++ b/ld/emultempl/pep.em
@@ -1866,7 +1866,8 @@ gld_${EMULATION_NAME}_finish (void)
     {
       pep_dll_fill_sections (link_info.output_bfd, &link_info);
       if (command_line.out_implib_filename
-          && pep_def_file->num_exports != 0)
+          && (pep_def_file->num_exports != 0
+              || bfd_link_pic (&link_info)))
 	pep_dll_generate_implib (pep_def_file,
 				 command_line.out_implib_filename, &link_info);
     }
-- 
2.28.0.windows.1

